src/js/components/DataTable/Header.js

```diff
@@ -91,8 +128,21 @@ const Header = ({
             verticalAlign,
             size,
           }) => {
-            let content =
-              typeof header === 'string' ? <Text>{header}</Text> : header;
+            // don't apply background or border on other contents of the cell
+            const contentStyles = { ...theme.dataTable.header };
+            delete contentStyles.background;
+            delete contentStyles.border;
```

Comment:
Shouldn't we remove `font` and `color` here too?

Reply:
I'm wondering if we should have a utility function inside this file that separates the `theme.dataTable.header` object into three objects: ` const [cellProps, layoutProps, textProps] = extricateThemeProps(theme);`? Then, `StyledHeaderCellButton` could use `layoutProps` and be sure it is using the same as the content `Box`. Just a thought.

Reply:
I like the idea of a utility function in this file. let me work on that

Reply:
Added `separateThemeProps` utility function in a recent commit that returns an array `[cellProps, layoutProps, textProps]` -- I felt that this function name would be more intuitive, but I liked the structure you proposed.